### DISCUSSION TITLE
How to use withAuthUserSSR when redirecting from url link?

### DISCUSSION THREAD

**Discussion by radim10** [gladly-team/next-firebase-auth#341] [Q&A]
I have an issue with using withAuthUserSSR when redirecting from a link. Lets's say I have a link to my app - http://localhost:3000/redirect.  I have a standard withAuthUserSSR page, but when I visit the page from URL link (not directly from my app with Next router), it returns AuthUser properties as null (email,  null...), looks like the auth hasn't been initialized. But when I go to the page from location in my app with Next router, it returns AuthUser as it should (auth is initialized already). I wonder if there is any way to achieve that with a redirect.

**Comment by kmjennison** [comment]
Could you provide an example app or code?

`withAuthUserSSR` is compatible with Next's redirect property in [`getServerSideProps`](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering), so that should work.

**Reply by radim10** [reply]
Well, it turned out that the problem was caused by the Component in _app. I have conditional rendering in this folder for my app so I will have to refactor it a little bit. Thanks! :) Btw I love this package.

**Reply by kmjennison** [reply]
Glad you figured it out, and thanks!

**Reply by radim10** [reply]
I 'repaired' the issue in _app but I found that the problem is probably because of third-party API. I am integrating my app with Zoom API (videoconferencing) and I have this oauth link with redirect to my app  `https://zoom.us/oauth/authorize?&client_id=XX&redirect_uri=http://localhost:3000/redirect`. After redirecting to my app (given redirect URL) the AuthUser in withAuthUserSSR is null but after refreshing it is not because the auth was initialized. Is there any workaround in this specific use-case?

**Reply by kmjennison** [reply]
You could make the redirect destination page (I assume a login page or similar) also redirect back to the app when AuthUser is defined. Or, change your Zoom auth redirect URI to another page that uses client-side auth.
